pkg/workceptor/lock_test.go

```diff
@@ -62,15 +62,15 @@ func TestStatusFileLock(t *testing.T) {
 				}
 				fileHasExisted = true
 				if err != nil {
-					t.Fatalf("Error loading status file: %s", err)
+					t.Errorf("Error loading status file: %s", err)
 				}
 				detailIter, err := strconv.Atoi(sfd.Detail)
 				if err != nil {
-					t.Fatalf("Error converting status detail to int: %s", err)
+					t.Errorf("Error converting status detail to int: %s", err)
 				}
 				if detailIter >= 0 {
 					if int64(sfd.State) != sfd.StdoutSize || sfd.State != detailIter {
-						t.Fatal("Mismatched data in struct")
+						t.Error("Mismatched data in struct")
```

Comment:
Since `t.Fatal()` exits only the go routine, and not the test, instead of calling either `t.Fatal()` or `t.Error()` can the go func return an `error` value. Then, the calling function check for errors from any of the go routines, and if any exist then call `t.Fatal()`. Also, in any error condition do not wait on any WaitGroup, simply fail the test.

This prevents the `wg2.Wait()` from potentially waiting forever.